src/dbus_api.rs

```diff
@@ -478,7 +482,65 @@ fn list_cache_devs(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
 }
 
 fn remove_cache_devs(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
-    Ok(vec![m.msg.method_return().append3("/dbus/cache/path", 0, "Ok")])
+    let message: &Message = m.msg;
+    let mut iter = message.iter_init();
+
+    let devs: Array<(&str), _> = try!(iter.read::<Array<(&str), _>>()
+        .map_err(|_| MethodErr::invalid_arg(&0)));
+
+    let dbus_context = m.path.get_data();
+    let object_path = m.path.get_name();
+    let return_message = message.method_return();
+    let return_sig = "(qs)";
+    let default_return = MessageItem::Array(vec![], return_sig.into());
+
+    let pool_name = dbus_try!(
+        object_path_to_pool_name(dbus_context, &object_path.to_string());
+        default_return; return_message);
+
+    let mut b_engine = dbus_context.engine.borrow_mut();
+    let ref mut pool = engine_try!(b_engine.get_pool(&pool_name);
+                                   default_return;
+                                   return_message);
+
+    let ref mut list_rc = ErrorEnum::OK;
+    let mut vec = Vec::new();
+    let to_remove_devs = devs.map(|x| Path::new(x)).collect::<Vec<&Path>>();
+
+    for dev in to_remove_devs {
+        let result = pool.remove_cachedev(dev);
+        match result {
+            Ok(_) => {
+                let cache_devs_map = dbus_context.cache_devs.borrow_mut();
+                if let Some(remove_object_path) =
+                       cache_devs_map.get_by_second(&String::from(dev.to_str()
+                    .unwrap())) {
```

Comment:
Something like:

```
for dev in devs {
    let result = pool.remove_cachedev(Path::new(dev));

    ...
    ... cache_devs_map.get_by_second(&String::from(dev)) ...
```

seems better to me if possible because

* it gets rid of that suspicious looking unwrap() (I agree it ought to be safe)
* it means one conversion instead of two
* it isolates converting to Path to one place instead of making it part of the overall structure of the code
  (I like this, because I think the signature of ```remove_cachedev``` is definitely wrong)